### PR TITLE
Most similar bug

### LINKED ISSUE #4443: Prune vectors error
## How to reproduce the behaviour
Same issue as #4386, #4388 doesn't quite fix it, as the first element of the tuple here: https://github.com/explosion/spaCy/blob/2d249a9502bfc5d3d2111165672d964b1eebe35e/spacy/vectors.pyx#L340 still has the shape of the full array instead of the batch. Pretty sure it just needs to be turned into this:

```sorted_index = xp.arange(scores.shape[0])[:,None][i:i+batch_size],xp.argsort(scores[i:i+batch_size], axis=1)[:,::-1]```

Will open a PR in case you agree. I would also suggest not sorting if n=1 since its not necessary so I'll also add that in the PR.

## Info about spaCy

* **spaCy version:** 2.2.1 (master)
* **Platform:** Linux-4.4.0-112-generic-x86_64-with-debian-stretch-sid
* **Python version:** 3.6.7

### PR DESCRIPTION
**Pull request by danielkingai2**
## Description

Fixes #4443

In https://github.com/explosion/spaCy/blob/2d249a9502bfc5d3d2111165672d964b1eebe35e/spacy/vectors.pyx#L340, the first element of the tuple needs to have the correct shape relative to the batch size, so should be replaced by `sorted_index = xp.arange(scores.shape[0])[:,None][i:i+batch_size],xp.argsort(scores[i:i+batch_size], axis=1)[:,::-1]`. Also there is no need to run the sorting code if only one most similar vector is requested.

### Types of change

Bug fix

## Checklist

- [x] I have submitted the spaCy Contributor Agreement.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by svlandeg** [comment]
Thanks for the PR!
This might not be very straightforward - but is there a way to write a unit test / regression test to check for the correct behaviour ?

**Comment by danielkingai2** [comment]
@svlandeg Sure, added a little test. See if it makes sense/is in the right place.

**Comment by danielkingai2** [comment]
Any idea when this fix might make it into a released version? It is blocking us from updating [scispacy](https://github.com/allenai/scispacy) to be compatible with the latest version of spacy, thanks :)

File: spacy/vectors.pyx [hunk 1]
```diff
@@ -336,8 +336,8 @@ cdef class Vectors:
             best_rows[i:i+batch_size] = xp.argpartition(sims, -n, axis=1)[:,-n:]
             scores[i:i+batch_size] = xp.partition(sims, -n, axis=1)[:,-n:]
 
-            if sort:
-                sorted_index = xp.arange(scores.shape[0])[:,None],xp.argsort(scores[i:i+batch_size], axis=1)[:,::-1]
+            if sort and n > 1:
```
Comment (honnibal) [inline spacy/vectors.pyx]: Tiny nitpick, but house style is to use `>=` and `<`, to make off-by-one errors more visually obvious.

```suggestion
            if sort and n >= 2:
```

**Comment by honnibal** [comment]
Thanks!

**Pull request merged**